File: Source/SaveSystem2.0/SyncUtil/EncodingHex.cs
using System;
using System.Globalization;

public static class EncodingHex
{
    // Converts a number to a 32-bit hex string
    public static string To32BitHexString(int number)
    {
        return Convert.ToString(number, 16).PadLeft(8, '0').ToUpper();
    }


    // Converts a hex string to an integer
    public static int ToNumber(string someHexString)
    {
        try
        {
            return Convert.ToInt32(someHexString, 16);
        }
        catch
        {
            return 0;
        }
    }

}

File: src/SaveSystem2.0/SyncUtil/EncodingHex.ts
export class EncodingHex {
    // Converts a number to a 32-bit hex string
    public static To32BitHexString(number: number): string {
        return number.toString(16).padStart(8, '0').toUpperCase()
    }

    // Converts a hex string to an integer
    public static ToNumber(someHexString: string): number {
        try {
            return parseInt(someHexString, 16)
        } catch {
            return 0
        }
    }
}
